graphics/include/ignition/common/ColladaExporter.hh

```diff
@@ -32,6 +33,41 @@ namespace ignition
 {
   namespace common
   {
+    /// \brief This struct contains light data specifically for collada export
+    /// Defaults set based on collada 1.4 specifications
+    struct ColladaLight
+    {
+      /// \brief Name of the light
+      std::string name;
+
+      /// \brief Type of the light. Either "point", "directional" or "spot"
+      std::string type;
+
+      /// \brief Light direction (directional/spot lights only)
+      math::Vector3d direction;
+
+      /// \brief Light position (non directional lights only)
+      math::Vector3d position;
```

Comment:
As I commented on https://github.com/ignitionrobotics/ign-gazebo/pull/912#discussion_r681903981, we should use the light's full `math::Pose3d` and apply a direction on top of that.

Now that this has been released, we can't extend this struct though. That's why we prefer full classes with getters and setters using the PIMPL pattern; they involve a lot more boilerplate, but at least they can be extended without breaking ABI.

I think the solution will have to be calculating the `direction` field based on the light's pose orientation together with the light's direction.

> /// \brief Light position (non directional lights only)

For directional lights, the position doesn't affect how they light up the scene, but it's often useful to set so we choose where the light's visual representation shows up on the GUI.

Reply:
Sorry for the delayed reply, was pushing some deadlines.

> Now that this has been released, we can't extend this struct though.

If I may ask, why so? Is there some technical reason?

This struct is specifically used for collada exporting - meaning if anyone decides to use it for something else they are taking on certain risks. I'm of the opinion that we should be able iterate quickly to what we want at an early stage when few are using it.

Reply:
> If I may ask, why so? Is there some technical reason?

ABI compatibility :confused: 

---

I think you may have solved the issue in https://github.com/ignitionrobotics/ign-gazebo/pull/912#discussion_r690976656